lib/ansible/plugins/filter/mathstuff.py

```diff
@@ -113,18 +118,53 @@ def human_readable(size, isbits=False, unit=None):
     ''' Return a human readable string '''
     try:
         return basic.bytes_to_human(size, isbits, unit)
-    except:
+    except Exception as e:
         raise errors.AnsibleFilterError("human_readable() can't interpret following string: %s" % size)
 
 
 def human_to_bytes(size, default_unit=None, isbits=False):
     ''' Return bytes count from a human readable string '''
     try:
         return basic.human_to_bytes(size, default_unit, isbits)
-    except:
+    except Exception as e:
         raise errors.AnsibleFilterError("human_to_bytes() can't interpret following string: %s" % size)
 
 
+def rekey_on_member(data, key):
+    """
+    Rekey a dict of dicts on another member
+
+    May also create a dict from a list of dicts.
+    """
+    new_obj = {}
+
+    if isinstance(data, collections.Mapping):
+        iterate_over = data.values()
+    elif isinstance(data, collections.Iterable) and not isinstance(data, (text_type, binary_type)):
+        iterate_over = data
+    else:
+        raise errors.AnsibleFilterError("Type is not a valid list, set, or dict")
+
+    for item in iterate_over:
+        if not isinstance(item, collections.Mapping):
+            raise errors.AnsibleFilterError("List item is not a valid dict")
+
+        try:
+            key_elem = item[key]
+        except KeyError:
+            raise errors.AnsibleFilterError("Key {0} was not found".format(key))
+        except Exception as e:
+            raise errors.AnsibleFilterError(to_native(e))
+
+        # TODO: We may want to add parameters to allow overwrite or constructing lists instead of erroring
+        if new_obj.get(key_elem, None):
```

Comment:
I think we should add an `allow_overwrite` or similar param.

Reply:
Done.